src/sdn_controller.py

```diff
@@ -55,10 +56,10 @@ def allocate(self):
         end_slot = self.spectrum_obj.spectrum_props.end_slot
         core_num = self.spectrum_obj.spectrum_props.core_num
         band = self.spectrum_obj.spectrum_props.curr_band
-        self.sdn_props.curr_band = band
 
-        if self.engine_props['guard_slots']:
-            end_slot = end_slot - 1
+        if self.engine_props['guard_slots'] != 0:
```

Comment:
I think I commented this before but we should definitely have a test for the guard slots if we don't already.